pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go

```diff
@@ -51,7 +52,31 @@ type ClusterDNS struct {
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// MultiClusterServiceDNSRecord
+// MultiClusterServiceDNSRecord defines a scheme of DNS name and subdomains that
+// should be programmed with endpoint information about a Service deployed in
+// multiple Kubernetes clusters. MultiClusterServiceDNSRecord is name-associated
+// with the Services it programs endpoint information for, meaning that a
+// MultiClusterServiceDNSRecord with name 'test-service' in namespace
+// 'test-namespace' expresses the intent to program DNS with information about
+// endpoints for the Kubernetes Service resources with the same name and
+// namespace in different clusters.
+//
+// MultiClusterServiceDNSRecord's spec defines elements of base DNS name that
+// contains all reachable endpoints for the Service in any cluster. This name is
+// of the form:
+//
+// (metadata.name).(metadata.namespace).(spec.federationName).svc
+//
+// For the example service mentioned earlier this would yield:
+//
+// test-service.test-namespace.(spec.federationName).svc
+//
+// Additionally, subdomains of this DNS name will be programmed for endpoints on
+// a region and zone basis:
+//
+// (status.DNS[*].region).test-service.test-namespace.(spec.federationName).svc
+// (status.DNS[*].zone).(status.DNS[*].region).test-service.test-namespace.(spec.federationName).svc
```

Comment:
Thanks @pmorie for this godoc.
This is slightly incorrect. There will be 3 names per cluster service instance.
and naming scheme is as follows:
1. Zone Level - (metadata.name).(metadata.namespace).(spec.federationName).svc.(status.DNS[\*].zone).(status.DNS[*].region).(spec.DNSSuffix)
2. Region Level - (metadata.name).(metadata.namespace).(spec.federationName).svc.(status.DNS[*].region).(spec.DNSSuffix)
3. Global Level - (metadata.name).(metadata.namespace).(spec.federationName).svc.(spec.DNSSuffix)